test/scripts/e2e-cp.sh

```diff
@@ -33,147 +33,152 @@ queries to the hostNetworked server pod on another node shall work for UDP|\
-	SKIPPED_TESTS+="|"
-  fi
-  SKIPPED_TESTS+="Node IP and MAC address migration"
+  skip "Node IP and MAC address migration"
 fi
 
 # Only run Multi node zones interconnect tests if they are explicitly requested
 MULTI_NODE_ZONES_TESTS="Multi node zones interconnect"
 if [[ "${WHAT}" != "${MULTI_NODE_ZONES_TESTS}"* ]]; then
-  if [ "$SKIPPED_TESTS" != "" ]; then
-	SKIPPED_TESTS+="|"
-  fi
-  SKIPPED_TESTS+="Multi node zones interconnect"
+  skip "Multi node zones interconnect"
 fi
 
 # Only run external gateway tests if they are explicitly requested
 EXTERNAL_GATEWAY_TESTS="External Gateway"
 if [[ "${WHAT}" != "${EXTERNAL_GATEWAY_TESTS}"* ]]; then
-  if [ "$SKIPPED_TESTS" != "" ]; then
-	SKIPPED_TESTS+="|"
-  fi
-  SKIPPED_TESTS+="External Gateway"
+  skip "External Gateway"
 fi
 
 # Only run kubevirt virtual machines tests if they are explicitly requested
 KV_LIVE_MIGRATION_TESTS="Kubevirt Virtual Machines"
 if [[ "${WHAT}" != "${KV_LIVE_MIGRATION_TESTS}"* ]]; then
-  if [ "$SKIPPED_TESTS" != "" ]; then
-	SKIPPED_TESTS+="|"
-  fi
-  SKIPPED_TESTS+=$KV_LIVE_MIGRATION_TESTS
+  skip $KV_LIVE_MIGRATION_TESTS
 fi
 
 # Only run network segmentation tests if they are explicitly requested
 NETWORK_SEGMENTATION_TESTS="Network Segmentation"
 if [[ "${WHAT}" != "${NETWORK_SEGMENTATION_TESTS}"* ]]; then
-  if [ "$SKIPPED_TESTS" != "" ]; then
-	SKIPPED_TESTS+="|"
-  fi
-  SKIPPED_TESTS+=$NETWORK_SEGMENTATION_TESTS
+  skip $NETWORK_SEGMENTATION_TESTS
 fi
 
-# Only run bgp tests if they are explicitly requested
 BGP_TESTS="BGP"
-if [[ "${WHAT}" != "${BGP_TESTS}"* ]]; then
-  if [ "$SKIPPED_TESTS" != "" ]; then
-	SKIPPED_TESTS+="|"
+if [ "$ENABLE_ROUTE_ADVERTISEMENTS" != true ]; then
+  skip $BGP_TESTS
+else
+  if [ "$ADVERTISE_DEFAULT_NETWORK" = true ]; then
+    # Some test don't work when the default network is advertised, either because
+    # the configuration that the test excercises does not make sense for an advertised network, or
+    # there is some bug or functional gap
+    # call out case by case
+
+    # pod reached from default network through secondary interface, asymetric, configuration does not make sense
+    # TODO: perhaps the secondary network attached pods should not be attached to default network
+    skip "Multi Homing A single pod with an OVN-K secondary network attached to a localnet network mapped to breth0 can be reached by a client pod in the default network on the same node"
+    skip "Multi Homing A single pod with an OVN-K secondary network attached to a localnet network mapped to breth0 can be reached by a client pod in the default network on a different node"
+  
+    # these tests require metallb but the configuration we do for it is not compatible with the configuration we do to advertise the default network
+    # TODO: consolidate configuration
+    skip "Load Balancer Service Tests with MetalLB"
+    skip "EgressService"
+
+    # tests that specifically expect the node SNAT to happen
```

Comment:
oh for EIP.. the ones that expect the node SNAT to happen is cause of the flow in br-ex being lower prio i assume?

Reply:
I am not sure we we are having a two sided discussion for the same topic, in that case we should continue only one of them.

Note that all EIPs tests toggle EIPs on and off, and when they are off they check that the traffic is SNATed to node IP. So not all the tests I am skipping here are just intra-cluster EIP tests, it's just a generic thing that test might at some point, for whatever reason, expect an SNAT that doesn't happen if the network is advertised.